docs/quickstart.fsx

```diff
@@ -0,0 +1,229 @@
+
+See the [DiffSharp.Data](/reference/diffsharp-data.html) namespace for the full API reference.
+
+### Datasets
+
+DiffSharp has ready-to-use types that cover main datasets typically used in machine learning, such as `cref:T:DiffSharp.Data.MNIST`, `cref:T:DiffSharp.Data.CIFAR10`, `cref:T:DiffSharp.Data.CIFAR100`, and also more generic dataset types such as `cref:T:DiffSharp.Data.TensorDataset` or `cref:T:DiffSharp.Data.ImageDataset`.
+
+The following loads the [MNIST](http://yann.lecun.com/exdb/mnist/) dataset and shows one image entry and the corresponding label.
+*)
+
+open DiffSharp
+open DiffSharp.Data
+
+// First ten images in MNIST training set
+let dataset = MNIST("../data", train=true, transform=id, n=10)
+
+// Inspect a single image and label
+let data, label = dataset.[7]
+
+// Save image to file
+data.saveImage("test.png")
+
+(** *)
+
+(*** hide ***)
+pngToHtml "test.png"
+(*** include-it-raw ***)
+
+// Inspect data as ASCII and show label
+printfn "Data: %A\nLabel: %A" (data.toImageString()) label
+(*** include-output ***)
+
+(**
+ 
+### Data Loaders
+
+A data loader handles tasks such as constructing minibatches from an underlying dataset on-the-fly, shuffling the data, and moving the data tensors between devices. In the example below we show a single batch of six MNIST images and their corresponding classification labels.
+
+*)
+
+let loader = DataLoader(dataset, shuffle=true, batchSize=6)
+let batch, labels = loader.batch()
+
+printfn "%A\nLabels: %A" (batch.toImageString()) labels
+(*** include-output ***)
+
+(**
+
+In practice a data loader would be typically used to iterate over all minibatches in a given dataset in order to feed each minibatch through a machine learning model. One full iteration over the dataset would be called an "epoch". Typically you would perform multiple such epochs of iterations during the training of a model.
+
+*)
+let epochs = 20
+
+for epoch = 1 to epochs do
+    for i, data, labels in loader.epoch() do
+        printfn "Epoch %A, minibatch %A" epoch (i+1)
+        // Process the minibatch
+        // ...
+(**
+
+## Models
+
+Many machine learning models are differentiable functions whose parameters can be tuned via [gradient-based optimization](https://en.wikipedia.org/wiki/Gradient_descent), finding an optimum for an objective function that quantifies the fit of the model to a given set of data. These models are typically built as compositions non-linear functions and ready-to-use building blocks such as linear, recurrent, and convolutional layers.
+
+DiffSharp provides the most commonly used model building blocks including convolutions, transposed convolutions, batch normalization, dropout, recurrent and other architectures.
+
```

Comment:
I think add a comment that in DiffSharp these parameters are tensors that may themselves be subject to differentiation and gradient-taking, rather than being floating-point parameters.

Reply:
The "parameters" in line 112 refers to tensors that are the parameters of neural networks. I think you're talking about the hyperparameters of the optimization algorithms, which we should talk about in the way you're suggesting. I think we do that elsewhere in this page.

Reply:
Yes that's right